File: findMissingArticleDates.ts
import { existsSync } from "fs";
import { join } from "path";

const startDate = new Date("2025-09-17");
const endDate = new Date(); // Today

const missingDates: string[] = [];

let date = new Date(startDate);
while (date <= endDate) {
  const dateStr = date.toISOString().split("T")[0];

  if (!existsSync(join("data", "articles", dateStr))) {
    missingDates.push(dateStr);
  }

  date.setDate(date.getDate() + 1);
}

if (missingDates.length === 0) {
  console.log("No missing dates found!");
} else {
  console.log(`Found ${missingDates.length} missing date(s):\n`);
  missingDates.forEach((date) => console.log(date));
}

File: scripts/find_missing_article_dates.py
#!/usr/bin/env -S uv run
# /// script
# requires-python = ">=3.12"
# ///

from datetime import date, timedelta
from pathlib import Path

start_date = date(2025, 9, 17)
end_date = date.today()

missing_dates: list[str] = []
d = start_date
while d <= end_date:
    date_str = d.isoformat()
    if not Path("data", "articles", date_str).exists():
        missing_dates.append(date_str)
    d += timedelta(days=1)

if not missing_dates:
    print("No missing dates found!")
else:
    print(f"Found {len(missing_dates)} missing date(s):\n")
    for date_str in missing_dates:
        print(date_str)
